packages/hooks/src/hooks.spec.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect } from 'react';
+import { act, renderHook } from 'react-hooks-testing-library';
+import { render, cleanup } from 'react-testing-library';
+import {
+  useDocumentEventListener,
+  useElementNode,
+  useViewportSize,
+} from './index';
+
+afterAll(cleanup);
+
+describe('packages/hooks', () => {
+  describe('useDocumentEventListener', () => {
+    test('event callback should fire when enabled is true', () => {
+      const eventCallback = jest.fn();
+
+      renderHook(() => useDocumentEventListener('click', eventCallback));
+
+      act(() => {
+        document.body.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+      });
+
+      expect(eventCallback).toHaveBeenCalledTimes(1);
+    });
+
+    test('event callback should not fire when enabled is false', () => {
+      const eventCallback = jest.fn();
+
+      renderHook(() =>
+        useDocumentEventListener({
+          type: 'click',
+          eventCallback,
+          enabled: false,
+        }),
+      );
+
+      act(() => {
+        document.body.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+      });
+
+      expect(eventCallback).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  // Difficult to test a hook that measures changes to the DOM without having access to the DOM
+  describe.skip('useMutationObserver', () => {}); //eslint-disable-line jest/no-disabled-tests
+
+  describe('useElementNode', () => {
+    let count = 0;
+
+    function TestUseElementNode() {
+      const [refEl, setRefEl] = useElementNode();
+      useEffect(() => {
+        ++count;
+      }, [refEl]);
+      return <div ref={setRefEl} />;
+    }
+
+    test('it gets called twice', () => {
+      act(() => {
+        render(<TestUseElementNode />);
+      });
+
+      // without setTimeout, test ran before final render
```

Comment:
This feels wrong – the docs don't seem to imply this executes asynchronously. timeouts with magic numbers (like `10`) are brittle, and may lead to false negatives down the line...

Reply:
I think because it renders twice it takes longer, I'm not sure how to properly accommodate this in the tests though.

Reply:
I'd do some research – I think it's important we address this properly to make sure our tests are reliable.